docs/fornax_test/830_release_testplan.md

```diff
@@ -0,0 +1,584 @@
+
+
+## Overview of Test Coverage
+
+The test cases can be classified into three groups. 
+
+**Group I: EdgeCluster Connection**
+
+Test Case 1: Register Hierarchical Edge Clusters with Cloud
+
+Test Case 3: An edge cluster re-connects to the cloud
+
+Test Case 5: An edge cluster continue to work when disconnected from the cloud
+
+Test Case 6: An edge cluster conintues to monintor its sub-edge-clusters when disconnected from cloud
+
+
+
+**Group II: Workload Managment**
+
+Test Case 4: Deploy workload (deployment) to edge clusters using mission
+
+Test Case 7: Deploy workload (job) to edge clusters using mission
+
+Test Case 8: Check mission status using command "kubectl get missions" 
+
+Test Case 9: Update workload to edge clusters using mission
+
+Test Case 10: The mission status of a cluster is "cluster unreachable" if cluster disconnected
+
+Test Case 11: Delete workload to edge clusters using mission
+
+Test Case 12: Deploy workload to an edge cluster with specific name
+
+Test Case 13: Deploy workload to selective edge clusters with given labels
+
+Test Case 15: An edge cluster picks up the new mission added during its disconnection when reconnected to the cloud
+
+Test Case 16: An edge cluster updates the mission updated during its disconnection when reconnected to the cloud
+
+Test Case 17: An edge cluster deletes the mission deleted during its disconnection when reconnected to the cloud
+
+Test Case 18: The mission content deleted in edge cluster will be re-instated
+
+Test Case 19: The mission content changed in edge cluster will be reverted automatically
+
+
+**Group III: Montioring EdgeCluster & Mission States**
+
+Test Case 2: Check edge cluster status when disconnected from the cloud 
+
+Test Case 14: Check edge cluster status when connected from the cloud 
+
+Test Case 20: The status of edge cluster is "Unhealthy" if the clusterd is connected but the underlying cluster is unreachable
+
+
+## Test Cases
+
+**Note: By default, the commands in this doc are run from the root directory of the fornax repo. If not specified, the command is run on the operating machine of Cluster A, which will be referred as "root operator machine" in the rest of this document.**
+
+
+**Test Case 1: Register Cascading Edge Clusters with Cloud**
+
+Step 1: start kubeedge cloudcore in the root operator machine, using command:
+```
+_output/local/bin/cloudcore
+```
+
+Step 2: Run the following command in the root operator machine to verify that **NO** edge cluster is registered.
+```
+kubectl get edgeclusters
+```
+
+Step 3: in Cluster B machine, start the edgecore in edge-cluster mode, with command 
+```
+_output/local/bin/edgecore --edgecluster
+```
+
+
+Step 4: wait 10-20 seconds, in the root operator machine, verify that the edge-cluster B shows up in the command ouput of 
+```
+kubectl get edgeclusters
+```
+
+Step 5: in cluster B machine, start cloudcore
```

Comment:
How's the certificate/token set up for this?

Reply:
this cert/token is taken care of in the "test cluster setup". Especially, it is done by the tool update_edgecore_config.sh, which is created in this PR.